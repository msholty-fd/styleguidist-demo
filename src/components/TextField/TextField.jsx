import React, { PropTypes } from 'react';
import TextField from 'material-ui/TextField';

import input from 'styleguide/components/input';
import hint from 'styleguide/components/hint';
import textFieldStyles from 'styleguide/components/textField';

const propTypes = {
  ...TextField.propTypes,
  children: PropTypes.array,
};

const getStyles = props => ({
  hint: Object.assign({}, hint, props.hintStyle),
  input: Object.assign({}, input, props.inputStyle),
  underlineFocus: Object.assign({}, textFieldStyles.underlineFocus, props.underlineFocusStyle),
  underline: Object.assign({}, textFieldStyles.underline, props.underlineStyle),
  base: Object.assign({}, textFieldStyles.base, props.base),
});

const SharpspringTextField = (props) => {
  const styles = getStyles(props);

  return (
    <TextField
      {...props}
      inputStyle={styles.input}
      hintStyle={styles.hint}
      underlineStyle={styles.underline}
      underlineFocusStyle={styles.underlineFocus}
      style={styles.base}
    >
      {props.children}
    </TextField>
  );
};

SharpspringTextField.propTypes = propTypes;

export default SharpspringTextField;
