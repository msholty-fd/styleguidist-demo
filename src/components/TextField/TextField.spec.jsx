import React from 'react'
import renderer from 'react-test-renderer'
import TextField from './TextField'
import { describe, it, expect } from 'global'

const component = <TextField name='jest' />

describe('<TextField />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(component)
    const tree = renderedComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
