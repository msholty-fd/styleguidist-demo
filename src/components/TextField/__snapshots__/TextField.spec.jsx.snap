// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should match the snapshot 1`] = `
<div
  className={undefined}
  style={
    Object {
      "backgroundColor": "transparent",
      "boxSizing": "border-box",
      "cursor": "auto",
      "display": "inline-block",
      "fontFamily": "\\"Lato\\", Helvetica, Arial, sans-serif",
      "fontSize": 16,
      "height": "40px",
      "lineHeight": "24px",
      "muiPrepared": true,
      "position": "relative",
      "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": 256,
    }
  }
>
  <input
    disabled={false}
    id="jest-undefined-undefined-6667"
    name="jest"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "WebkitTextFillColor": "#555555",
        "appearance": "textfield",
        "backgroundColor": "rgba(0,0,0,0)",
        "border": "2px solid #C3C8C8",
        "boxSizing": "border-box",
        "color": "#555555",
        "cursor": "inherit",
        "font": "inherit",
        "fontSize": 14,
        "height": "40px",
        "lineHeight": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "paddingLeft": 5,
        "paddingRight": 0,
        "position": "relative",
        "width": "100%",
      }
    }
    type="text"
  />
  <div>
    <hr
      aria-hidden="true"
      style={
        Object {
          "border": "none",
          "borderBottom": "solid 1px",
          "borderColor": "#e0e0e0",
          "borderLeft": "none",
          "borderRight": "none",
          "borderTop": "none",
          "bottom": 8,
          "boxSizing": "content-box",
          "margin": 0,
          "muiPrepared": true,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <hr
      aria-hidden="true"
      style={
        Object {
          "border": "2px solid #40acc3",
          "borderBottom": "solid 2px",
          "borderColor": "#00bcd4",
          "borderLeft": "none",
          "borderRadius": "3px",
          "borderRight": "none",
          "borderTop": "none",
          "bottom": 8,
          "boxSizing": "border-box",
          "height": "40px",
          "margin": 0,
          "muiPrepared": true,
          "position": "absolute",
          "top": 0,
          "transform": "scaleX(0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
          "zIndex": "4",
        }
      }
    />
  </div>
</div>
`;
